src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerConfigTests.cs

```diff
@@ -2101,6 +2100,75 @@ public void GlobalConfigOptionsAreEmptyWhenNoGlobalConfig()
             Assert.Empty(globalOptions.TreeOptions);
         }
 
+        [Theory]
+        [InlineData("/path/to/file.cs", true)]
+        [InlineData("file.cs", false)]
+        [InlineData("../file.cs", false)]
+        [InlineData("**", false)]
+        [InlineData("*.cs", false)]
+        [InlineData("?abc.cs", false)]
+        [InlineData("/path/to/**", false)]
+        [InlineData("/path/[a]/to/*.cs", false)]
+        [InlineData("/path{", false)]
+        [InlineData("/path}", false)]
+        [InlineData("/path?", false)]
+        [InlineData("/path,", false)]
+        [InlineData("/path\"", true)]
+        [InlineData("/path\\", false)] //editorconfig sees a single escape character (special)
+        [InlineData("/path\\\\", true)] //editorconfig sees an escaped backslash
+        [InlineData("//path", true)]
+        [InlineData("//", true)]
+        [InlineData("\\", false)] //invalid: editorconfig sees a single escape character
+        [InlineData("\\\\", false)] //invalid: editorconfig sees an escaped, literal backslash
+        [InlineData("/\\{\\}\\,\\[\\]\\*", true)]
+        [InlineData("c:\\my\\file.cs", false)] // invalid: editorconfig sees a single file called 'c:(\m)y(\f)ile.cs' (i.e. \m and \f are escape chars)
+        [InlineData("\\my\\file.cs", false)] // invalid: editorconfig sees a single file called '(\m)y(\f)ile.cs' 
+        [InlineData("\\\\my\\\\file.cs", false)] // invalid: editorconfig sees a single file called '\my\file.cs' with literal backslashes
+        [InlineData("\\\\\\\\my\\\\file.cs", false)] // invalid: editorconfig sees a single file called '\\my\file.cs' not a UNC path
+        [InlineData("//server/file.cs", true)]
+        [InlineData("//server\\file.cs", true)]
+        [InlineData("\\/file.cs", true)] // allow escaped chars
+        [InlineData("<>a??/b.cs", false)]
+        [InlineData(".", false)]
+        [InlineData("/", true)]
+        [InlineData("", true)] // only true becuase [] isn't a valid editorconfig section name either and thus never gets parsed
+        public void GlobalConfigIssuesWarningWithInvalidSectionNames(string sectionName, bool isValid)
+        {
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse($@"
+is_global = true
+[{sectionName}]
+", "/.editorconfig"));
+
+            _ = AnalyzerConfigSet.Create(configs, out var diagnostics);
+            configs.Free();
+
+            if (isValid)
+            {
+                diagnostics.Verify();
+            }
+            else
+            {
+                diagnostics.Verify(
+                    Diagnostic("InvalidGlobalSectionName", isSuppressed: false).WithArguments(sectionName, "/.editorconfig").WithLocation(1, 1)
+                    );
+            }
+        }
+
+        [ConditionalTheoryAttribute(typeof(WindowsOnly))]
```

Comment:
>ConditionalTheoryAttribute(typeof(WindowsOnly)) [](start = 9, length = 47)

Should we test these same cases for other platforms as well, even though the `isValid` result might be different? If so, perhaps the test could cover both cases.
```C#
[Theory]
[InlineData("c:/myfile.cs", true, false)]
...
public void GlobalConfigIssuesWarning(string sectionName, bool isValidWindows, bool isValidOther)
{
    GlobalConfigIssuesWarningWithInvalidSectionNames(sectionName,
        ExecutionConditionUtil.IsWindows ? isValidWindows : isValidOther);
}
```